co-circom/circom-mpc-vm/src/mpc/rep3.rs

```diff
@@ -571,6 +572,38 @@ impl<F: PrimeField, N: Rep3Network> VmCircomWitnessExtension<F>
             }
         }
     }
+
+    fn addbits(
+        &mut self,
+        a: Vec<Self::VmType>,
+        b: Vec<Self::VmType>,
+    ) -> eyre::Result<(Vec<Self::VmType>, Self::VmType)> {
+        assert!(a.len() == b.len());
+        let bitlen = a.len();
+        let a = a.into_iter().map(|x| match x {
+            Rep3VmType::Public(x) => promote_to_trivial_share(self.io_context0.id, x),
+            Rep3VmType::Arithmetic(x) => x,
+        });
+        let b = b.into_iter().map(|x| match x {
+            Rep3VmType::Public(x) => promote_to_trivial_share(self.io_context0.id, x),
+            Rep3VmType::Arithmetic(x) => x,
+        });
+        let two = F::one() + F::one();
+
+        let a_sum = a.fold(Rep3PrimeFieldShare::zero_share(), |acc, x| acc * two + x);
+        let b_sum = b.fold(Rep3PrimeFieldShare::zero_share(), |acc, x| acc * two + x);
+
+        let sum = a_sum + b_sum;
```

Comment:
are we sure this never overflows the field? otherwise we can not get the carry and the result might be wrong